planetiler-core/src/main/java/com/onthegomap/planetiler/util/Format.java

```diff
@@ -17,7 +17,15 @@
 public class Format {
 
   public static final Locale DEFAULT_LOCALE = Locale.getDefault(Locale.Category.FORMAT);
-  public static final ConcurrentMap<Locale, Format> instances = new ConcurrentHashMap<>();
+
+  // `NumberFormat` instances are not thread safe, so we need to wrap them inside a `ThreadLocal`.
+  //
+  // Ignore warnings about not removing thread local values since planetiler uses dedicated worker threads that release
+  // values when a task is finished and are not re-used.
+  @SuppressWarnings("java:S5164")
+  private static final ThreadLocal<ConcurrentMap<Locale, Format>> instances = ThreadLocal.withInitial(
+    ConcurrentHashMap::new);
+
   private final NumberFormat pf;
   private final NumberFormat nf;
   private final NumberFormat intF;
```

Comment:
I think we want to do these 3 as well.

Reply:
Ah, you're right. I was hoping there wasn't a case where one thread constructs a `Format` instance and then shares it to another thread, but seems that very likely is the case with `ProgressLoggers`